provisioning_test.go

```diff
@@ -475,21 +508,30 @@ func TestRecreateSRK(t *testing.T) {
 
 			lockoutAuth := []byte("1234")
 
-			if err := ProvisionTPM(tpm, ProvisionModeFull, lockoutAuth); err != nil {
-				t.Fatalf("ProvisionTPM failed: %v", err)
+			if err := tpm.EnsureProvisioned(ProvisionModeFull, lockoutAuth); err != nil {
+				t.Fatalf("EnsureProvisioned failed: %v", err)
 			}
 
 			srk, err := tpm.CreateResourceContextFromTPM(tcg.SRKHandle)
 			if err != nil {
 				t.Fatalf("No SRK context: %v", err)
 			}
+			expectedName := srk.Name()
 
 			if _, err := tpm.EvictControl(tpm.OwnerHandleContext(), srk, srk.Handle(), nil); err != nil {
 				t.Errorf("EvictControl failed: %v", err)
 			}
 
-			if err := ProvisionTPM(tpm, data.mode, lockoutAuth); err != nil {
-				t.Fatalf("ProvisionTPM failed: %v", err)
+			if err := tpm.EnsureProvisioned(data.mode, lockoutAuth); err != nil {
+				t.Fatalf("EnsureProvisioned failed: %v", err)
+			}
+
+			srk, err = tpm.CreateResourceContextFromTPM(tcg.SRKHandle)
```

Comment:
this is an additional test now?

Reply:
It is - it verifies that the TPM hasn't been cleared by making sure that the new primary key is identical to the original one.